### PR TITLE
Explain the proper way to incude scripts and stylesheets in widgets

### PR DESCRIPTION
**Pull request by Hyzual**
New widgets should use those mechanisms. They help us avoid including 17
scripts and stylesheets on an empty dashboard and therefore will improve
the dashboard pages' performance.

### CODE REVIEW

File: languages/en/developer-guide/back-end/widgets.rst [hunk 1]
```diff
@@ -93,4 +93,45 @@ The widget must be referenced in 2 places:
...
\ No newline at end of file
+You can now add a new widget on your personal page, it will display a simple message within.
+
+Javascript and CSS dependencies
+-------------------------------
+
+It's becoming common to have complicated widgets with lots of user interaction. When you need to include :ref:`Javascript files<dev-javascript>` or :ref:`CSS files<dev-css>`, you'll need to override those two functions:
+
+.. code-block:: php
+
```
Comment (gorkat) [inline languages/en/developer-guide/back-end/widgets.rst]: can you add in which file this code should be included?

File: languages/en/developer-guide/back-end/widgets.rst [hunk 2]
```diff
@@ -93,4 +93,45 @@ The widget must be referenced in 2 places:
...
+            array('snippet' => 'window.CKEDITOR_BASEPATH = "' . $ckeditor_path . '";'),
+            array('file' => $ckeditor_path . 'ckeditor.js'),
+            array('file' => $kanban_include_assets->getFileURL('kanban.js')),
+        );
+    }
+
+The previous code block shows an example with the Kanban widget. It returns an array of arrays. Each array must have either a 'file' key or a 'snippet' key. 'file' keys *can* have a 'unique-name'. 'unique-name' files will be included only once for all widgets present on the page.
+
+.. code-block:: php
+
```
Comment (gorkat) [inline languages/en/developer-guide/back-end/widgets.rst]: Same

File: languages/en/developer-guide/back-end/widgets.rst (hunk 1 above)
Comment (Hyzual) [reply languages/en/developer-guide/back-end/widgets.rst]: Done

File: languages/en/developer-guide/back-end/widgets.rst (hunk 2 above)
Comment (Hyzual) [reply languages/en/developer-guide/back-end/widgets.rst]: Done

**Pull request merged**